frontend/js/pages/BattleDetail.js

```diff
@@ -1,84 +1,103 @@
   }, []);
+
+  const { battle, user } = props;
+
   if (!battle) {
     return (
-      <MessageBanner
-        content={
-          <div className="body">
-            <div className="teste">
-              <h1>The battle you are looking for does not exist.</h1>
-            </div>
-          </div>
-        }
-      />
+      <Wrapper>
+        <Title>The battle you are looking for does not exist.</Title>
+      </Wrapper>
     );
   }
-  if (!battle.winner || !battle.teams) {
-    return (
-      <MessageBanner
-        content={
+
+  const { currentUserTeam, opponentUserTeam } = showTeams(battle, user);
+
+  return (
+    <Wrapper>
+      {user.email !== battle.creator.email && user !== battle.opponent.email ? (
+        <Text>You do not have acess to this battle</Text>
+      ) : (
+        <div>
+          <Title>Battle result!</Title>
+          {battle.winner ? (
+            <Text>The winner is {battle.winner ? battle.winner.email : ''}</Text>
+          ) : (
+            ''
+          )}
           <div>
-            <h1>Battle result!</h1>
             <div>
-              <h1>The battle is not over yet!</h1>
-              <p>{battle.creator.email} team:</p>
-              <div>
-                <a href={Urls['battle-list']()}>Back</a>
-              </div>
+              <p>Your team:</p>
+              {currentUserTeam.pokemons.length === 0 ? (
+                <div>
+                  <p>You have not chosen your pokemon yet.</p>
+                  <a href={getTeamUrl(currentUserTeam.id)} role="button">
+                    Edit your team
+                  </a>
+                </div>
+              ) : (
+                <TeamCard pokemons={currentUserTeam.pokemons} />
+              )}
             </div>
-          </div>
-        }
-      />
-    );
-  }
-  return (
-    <div className="body">
-      <div className="container1" id="home">
-        <h1>Welcome to Poke Battle!</h1>
-        <h2>Choose your pokemons and fight!</h2>
-        <div className="container2">
-          <img
-            alt="pokemon_img"
-            height="200px"
-            src="https://static.tvtropes.org/pmwiki/pub/images/pokemon_350_210.png"
-            width="400px"
-          />
-
-          <div className="teste">
-            <h1>Result</h1>
             <div>
-              <h3>{battle.winner.email} won!</h3>
-              <TeamTable battleTeam={battle.teams[0]} battleTrainer={battle.creator} />
-              <TeamTable battleTeam={battle.teams[1]} battleTrainer={battle.opponent} />
+              <p>Your opponent team:</p>
+              {battle.winner ? (
+                <TeamCard pokemons={opponentUserTeam.pokemons} />
+              ) : (
+                <p>The battle is not over yet.</p>
+              )}
             </div>
-            <a className="button_next" href={Urls['battle-list']()}>
-              Back
-            </a>
           </div>
+          <a href={getTeamUrl(getBattleListPage)} role="button">
```

Comment:
This is not making much sense to me. Since this function receives an `id` and you are passing a function `getBattleListPage`.

Reply:
Please remove this and just pass the URL: `Urls.team_create(id)`